src/controllers/sessionController.ts

```diff
@@ -23,4 +31,23 @@ export default class SessionController {
         walletToken: user.wallet_token
       })
     );
+  public readonly listSessions = (
+    req: express.Request
+  ): Promise<
+    // tslint:disable-next-line: max-union-size
+    | IResponseErrorNotFound
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseSuccessJson<SessionsList>
+  > =>
+    withUserFromRequest(req, async user => {
+      const sessionsList = await this.sessionStorage.listUserSessions(user);
+      if (isLeft(sessionsList)) {
+        return ResponseErrorInternal(sessionsList.value.message);
+      }
+      if (sessionsList.value.sessions.length === 0) {
+        return ResponseErrorNotFound("Not Found", "User sessions not found");
```

Comment:
I would return a 500 here as this is a scenario that should never happen, the logged in user always has *at least* the current user session

Reply:
fixed in 3b7b721